File: firebase/functions/lib/admin/payments/stripe_connected_account_webhooks.dart
import 'package:firebase_functions_interop/firebase_functions_interop.dart';
import 'abstract_stripe_webhooks.dart';
import '../../utils/utils.dart';

class StripeConnectedAccountWebhooks extends AbstractStripeWebhooks {
  StripeConnectedAccountWebhooks() : super('StripeConnectedAccountWebhooks');

  @override
  String getKey() {
    return functions.config.get('stripe.connected_account_webhook_key')
        as String;
  }

  @override
  Future<String> action(JsonMap request) async {
    final type = request.json['type'];
    if (type == 'account.updated') {
      await handleAccountUpdated(request);
    } else {
      print('Unknown Stripe event type: ${request.json['type']}');
    }
    return Future.value('');
  }
}

File: firebase/functions/src/admin/payments/stripe_connected_account_webhooks.ts
import * as functions from 'firebase-functions';
import { AbstractStripeWebhooks } from './abstract_stripe_webhooks';

type JsonMap = Record<string, unknown>;

export class StripeConnectedAccountWebhooks extends AbstractStripeWebhooks {
  constructor() {
    super('StripeConnectedAccountWebhooks');
  }

  getKey(): string {
    return functions.config().stripe?.connected_account_webhook_key as string;
  }

  async action(request: JsonMap): Promise<string> {
    const type = request['type'] as string;
    if (type === 'account.updated') {
      await this.handleAccountUpdated(request);
    } else {
      console.log(`Unknown Stripe event type: ${type}`);
    }
    return '';
  }
}
